docarray/typing/tensor/ndarray.py

```diff
@@ -210,3 +210,7 @@ def get_comp_backend() -> Type['NumpyCompBackend']:
         from docarray.computation.numpy_backend import NumpyCompBackend
 
         return NumpyCompBackend
+
+    def __class_getitem__(cls, item: Any, *args, **kwargs):
+        # see here for mypy bug: https://github.com/python/mypy/issues/14123
+        return AbstractTensor.__class_getitem__.__func__(cls, item)  # type: ignore
```

Comment:
Encountered another mypy bug here

Reply:
why is the __func_ needed here ?

Reply:
otherwise `__class_getitem__` is bound to `AbstractTensor` and use that as the `cls` input, which will result in the parametrized type being a `AbstractTensor[shape]` instad of a `NdArray[shape]`.
`__func__` gets us the unbound function, so we can pass `cls` explicitly.